### PR TITLE
CostMatrix

### PR DESCRIPTION
**Pull request by dnesbitt61**
First version. Still working on termination criteria.

### CODE REVIEW

File: valhalla/thor/costmatrix.h [hunk 1]
```diff
@@ -0,0 +1,210 @@
+#ifndef VALHALLA_THOR_TIMEDISTANCEMATRIX2_H_
+#define VALHALLA_THOR_TIMEDISTANCEMATRIX2_H_
```
Comment (gknisely) [inline valhalla/thor/costmatrix.h]: #ifndef VALHALLA_THOR_TIMEDISTANCEMATRIX2_H_
#define VALHALLA_THOR_TIMEDISTANCEMATRIX2_H_
need to be renamed

File: valhalla/thor/costmatrix.h [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+  /**
+   * Form a time/distance matrix from the results.
+   * @return  Returns a time distance matrix among locations.
+   */
+  std::vector<TimeDistance> FormTimeDistanceMatrix();
+};
+
+}
+}
+
```
Comment (gknisely) [inline valhalla/thor/costmatrix.h]: same...rename

Comment (kevinkreiser) [inline valhalla/thor/costmatrix.h]: eagle :eyes:

**Comment by dgearhart** [comment]
:shipit:

**Pull request merged**